### ISSUE TITLE
Just a thought

### ISSUE THREAD

**Issue by datshi001** [gastush/ha-renac#11]
Hi @gastush ,
Just a thought...............
Should the device class 'Energy' rather than 'Measurement' be used for the generated power ententities? Might this allow them to be reported in the energy dashboard?

Currently I get the following 2 messages in the energy configuration set up:

_Unexpected device class_
The following entities do not have the expected device class:
sensor.renac_today_s_generated_power
_Last reset missing_
The following entities have state class 'measurement' but 'last_reset' is missing:
sensor.renac_today_s_generated_power

So I tried adding this to my config (which is a modified bit of code I found whilst trying to figure this out)  I don't know if it could be incorporated into the integration:
```
homeassistant:
  customize_glob:
    sensor.renac_today_s_generated_power:
      unit_of_measurement: "kWh"     
      device_class: energy
      state_class: total_increasing
      last_reset: "1970-01-01T00:00:00+00:00"
``` 
It seems to work...........(takes a couple of hours to get up and running on the energy dashboard)
Forgive my somewhat crude efforts but I'm still fairly close to the beginning of my learning curve here.:)

**Comment by gastush** [comment]
Makes sense indeed.... I've been using another way to feed the Energy dashboard:
```
  - platform: integration
    source: sensor.renac_generated_power
    name: renac_generated_energy
    unit_prefix: k
    round: 2
```
This was doing the job, but is somewhat a bit less accurate than the value from the Inverter.
I just made an update to use the proper Device and State class for all sensors. I'll see if it works well tomorrow (the inverter is offline at the time of writing). If it works well, I'll push a new version. The next version should be a bit more resistant to the expiry of the token.

Do note that 'total_increasing' and 'last_reset' are not compatible. So I used 'total' together with 'last_reset'. Will see if it works fine.

Lastly, I find the cloud APIs quite unreliable and moving too often, so I'm currently looking at how I can get the date straight from the Inverter through its Wifi module. This will require either direct Bluetooth connection, or a mix of Modbus and AT commands over Wifi. I'll keep you posted.
